controllers/windowsmachine_controller.go

```diff
@@ -420,6 +420,9 @@ func (r *WindowsMachineReconciler) addWorkerNode(ipAddress, instanceID, machineN
 		return errors.Wrapf(err, "failed to configure Windows VM %s", instanceID)
 	}
 	if err := nc.Configure(); err != nil {
+		if err := nc.MarkNotReady(); err != nil {
+			r.log.Info("Unable to mark node as NotReady", "error", err)
+		}
```

Comment:
I dont like this having to be called if Configure() fails.
nc.Configure() should be properly cleaning up after itself on failure, and not leave partially configured nodes dangling.

Reply:
Ideas for how to deal with this below